content/blog/get-started-with-prometheus-and-grafana-on-docker-with-hpe-storage-array-exporter.md

```diff
@@ -0,0 +1,162 @@
+
+```markdown
+docker network create prometheus
+```
+
+Note that all containers created below do not utilize any persistent storage at all. Any dashboards that have been created or metrics that have been gathered will not be accessible after a container restart.
+
+**Note:** This tutorial is purely for educational purposes and not intended for production deployments.
+
+# Run HPE Storage Array Exporter for Prometheus
+
+Let's begin by starting the HPE Storage Array Exporter for Prometheus container. It's a passive background process that responds to HTTP requests made by Prometheus. It also requires a backend configuration file. In the case of having multiple backends, one container and configuration file is needed for each one.
+
+Create a file named `my-array-1.yaml`:
+
+```yaml
+address: my-array-1
+username: admin
+password: admin
+```
+
+The `address` value can be an IP address or hostname resolvable in DNS. Then, run `docker`:
+
+```markdown
+docker run -d --rm --name my-array-1 \
+     -v $(pwd)/my-array-1.yaml:/etc/config/storage-system.yaml \
+     --network prometheus \
+     quay.io/hpestorage/array-exporter:v1.0.0-beta \
+     --log.path /var/log/hpe-array-exporter.log \
+     --accept-eula \
+     /etc/config/storage-system.yaml
+```
+
+By running the above command with the `--accept-eula` parameter, the end-user accepts the HPE End User License Agreement at <https://www.hpe.com/us/en/software/licensing.html>.
+
+The exporter is now up and running listening on port 8080. To ensure we can reach the metrics endpoint, run the following `docker` command:
+
+```markdown
+docker run --rm --network prometheus \
+    curlimages/curl -s http://my-array-1:8080/metrics
+```
+
+It should produce an output that enumerates the metric keys, value, type and brief help. Example:
+
+```markdown
+# HELP hpenimble_pool_snapshot_used_bytes Total logical size of all snapshots in a storage pool
+# TYPE hpenimble_pool_snapshot_used_bytes gauge
+hpenimble_pool_snapshot_used_bytes{pool="default"} 2.659803136e+09
+```
+
+Next, let's start Prometheus and scrape some metrics.
+
+# Run and Inspect Prometheus
+
+When Prometheus is deployed on Kubernetes, discovery of scrape targets may be configured to be performed dynamically. In this example we need to instruct Prometheus where to find our target. Create a configuration file named `prometheus.yml`:
+
+```yaml
+global:
+  scrape_interval: 15s
+  evaluation_interval: 15s
+
+scrape_configs:
+  - job_name: "my-array-1"
+    static_configs:
+      - targets: ["my-array-1:8080"]
+```
+
+Create the Prometheus container:
+
+```markdown
+docker run -d --rm --name prometheus-server \
+    --network prometheus \
+    -p 80:9090 \
+    -v $(pwd)/prometheus.yml:/etc/prometheus/prometheus.yml \
+    prom/prometheus:v2.30.3
+```
+
+In a few moments, the Prometheus server should be up and listening on the docker host and accessible from http://your.docker.host.example.com/. 
```

Comment:
Recommend hotlinking the URL and not spelling it out.
Also, "the docker host" should be "the Docker host"

Reply:
docker is a company, "docker" is a program.